### PR TITLE
feat(app-store-connect): Add manually sync button

### PR DESCRIPTION
**Pull request by priscilawebdev**
image

### CODE REVIEW

File: static/app/views/settings/projectDebugFiles/sources/customRepositories/index.tsx [hunk 1]
```diff
@@ -175,6 +176,21 @@ function CustomRepositories({
...
+  async function handleSyncRepositoryNow(repoId: CustomRepo['id']) {
+    try {
+      await api.requestPromise(
+        `/projects/${orgSlug}/${projSlug}/appstoreconnect/${repoId}/refresh/`,
+        {
+          method: 'POST',
+        }
+      );
+    } catch (error) {
+      const errorMessage = t('Unable to refresh repository. Try again in a few minutes.');
```
Comment (flub) [inline static/app/views/settings/projectDebugFiles/sources/customRepositories/index.tsx:188]: I find the "unable to fresh repository" part too vague: it does not say this is rate limited.  Maybe "Rate limit for refreshing repository exceeded. Try again in a few minutes." is better?

**Review by flub — COMMENTED** [review]
Looks good and works great.  My only reservation is that the "last refreshed" time does not change until the entire page is reloaded, which could be a bit confusing?  Only after a full page refresh do I get an indication the refresh worked.

One likely user scenario would be:
- User clicks "sync now"
- No feedback?  Weird.  Let's click it again.
- Oh no, rate limit exceeded?  What now?
- Maybe it did work?  Refresh page I guess.
- Oh, it did work!

Though I appreciate that this could be tricky to implement so happy to leave it unaddressed.  Or maybe there are other ways of improving this, like showing some kind of "ok" banner if the endpoint returned ok.

File: static/app/views/settings/projectDebugFiles/sources/customRepositories/index.tsx (hunk 1 above)
Comment (priscilawebdev) [reply static/app/views/settings/projectDebugFiles/sources/customRepositories/index.tsx:188]: sure thing. Great suggestion! thanks 😉

**Comment by flub** [comment]
> My only reservation is that the "last refreshed" time does not change until the entire page is reloaded

There's also the challenge that this "sync now" action actually happens asynchronous, so we can't really guarantee that the status has changed right after you got a 200 ok from the refresh endpoint it'll probably still show the same.  So perhaps showing the green success message in the bottom right is the better option.

**Review by Jesse-Box — APPROVED** [review]

File: static/app/views/settings/projectDebugFiles/sources/customRepositories/index.tsx [hunk 2]
```diff
@@ -175,6 +176,24 @@ function CustomRepositories({
...
 
+  async function handleSyncRepositoryNow(repoId: CustomRepo['id']) {
+    try {
+      await api.requestPromise(
+        `/projects/${orgSlug}/${projSlug}/appstoreconnect/${repoId}/refresh/`,
+        {
+          method: 'POST',
+        }
+      );
+      addSuccessMessage('Request to sync repository sent.');
```
Comment (matejminar) [inline static/app/views/settings/projectDebugFiles/sources/customRepositories/index.tsx:187]:
```suggestion
      addSuccessMessage(t('Request to sync repository sent.'));
```

**Review by matejminar — APPROVED** [review]
Code looks good to me.

File: static/app/views/settings/projectDebugFiles/sources/customRepositories/index.tsx [hunk 3]
```diff
@@ -184,8 +184,11 @@ function CustomRepositories({
           method: 'POST',
         }
       );
+      addSuccessMessage(t('Request to sync repository sent.'));
```
Comment (flub) [inline static/app/views/settings/projectDebugFiles/sources/customRepositories/index.tsx:187]: nit: I'd write "Repository sync started" but I'm not sure what tone of voice guidelines there are so maybe I'm talking nonsense.

**Review by flub — APPROVED** [review]
lgtm, thanks!

File: static/app/views/settings/projectDebugFiles/sources/customRepositories/index.tsx [hunk 4]
```diff
@@ -175,6 +176,24 @@ function CustomRepositories({
...
 
+  async function handleSyncRepositoryNow(repoId: CustomRepo['id']) {
+    try {
+      await api.requestPromise(
+        `/projects/${orgSlug}/${projSlug}/appstoreconnect/${repoId}/refresh/`,
+        {
+          method: 'POST',
+        }
+      );
+      addSuccessMessage(t('Request to sync repository sent.'));
```
Comment (priscilawebdev) [inline static/app/views/settings/projectDebugFiles/sources/customRepositories/index.tsx:187]:
```suggestion
      addSuccessMessage(t('Repository sync started.'));
```

File: tests/js/spec/views/settings/projectDebugFiles/customRepositories.spec.tsx [hunk 5]
```diff
@@ -307,4 +312,108 @@ describe('Custom Repositories', function () {
...
+
+      const syncNowButton = screen.getByRole('button', {name: 'Sync Now'});
+      expect(syncNowButton).toBeEnabled();
+
+      userEvent.click(syncNowButton);
+
+      await waitFor(() => expect(refreshMockSuccess).toHaveBeenCalledTimes(1));
+
+      expect(indicators.addSuccessMessage).toHaveBeenCalledWith(
+        'Request to sync repository sent.'
```
Comment (priscilawebdev) [inline tests/js/spec/views/settings/projectDebugFiles/customRepositories.spec.tsx:344]:
```suggestion
        'Repository sync started.'
```

**Pull request merged**